packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -370,6 +370,32 @@ export class AzureCommunicationCallAdapter<AgentType extends CallAgent | BetaTea
       }
 
       this.context.updateClientState(clientState);
+
+      /* @conditional-compile-remove(call-transfer) */
+      const acceptedTransferCallState = this.context.getState().acceptedTransferCallState;
+
+      /* @conditional-compile-remove(call-transfer) */
+      // If there has been an accepted transfer call that is now in the connected state, we should ensure we leave the
+      // current call. There is a possibility we accepted a transfer which the transferor thinks failed.
+      if (
+        acceptedTransferCallState &&
+        acceptedTransferCallState.state === 'Connected' &&
+        acceptedTransferCallState.id !== this.call?.id
+      ) {
+        const cAgent = callAgent as CallAgent;
+        const transferCall = cAgent.calls.find((call) => (call as Call).id === acceptedTransferCallState.id);
+        if (transferCall) {
+          const oldCall = this.call;
+          this.processNewCall(transferCall);
+          // Wait 2 seconds to allow transferor to recognize the transfer is complete and end the call. But if the old
```

Comment:
is there not another way we can wait for the state of the old call to change? if there is an issue with the transfer does the state not go to `none` or something?

Reply:
Unfortunately no the old call state is 'Connected' during the transfer and still 'Connected' when it fails.